Second week, section 1's pull request

DvdHalliday opened the pull request:
Exercise 1
![e1](https://user-images.githubusercontent.com/109089386/179070967-f078bb98-4bf7-4ea0-b675-34ec78ae1365.png)
Exercise 2
![e2](https://user-images.githubusercontent.com/109089386/179070973-8fe7b517-5ccf-4b79-8dc6-719eaa8c8bf5.png)
Exercise 3
![e3](https://user-images.githubusercontent.com/109089386/179070975-876744be-71ae-450f-b7ff-b61544530424.png)
Exercise 4
![e4](https://user-images.githubusercontent.com/109089386/179070976-268e486c-80d5-445b-8a44-4448d3a7dcf1.png)

DvdHalliday committed:
Exercise 1

Working Exercise 1 upload

DvdHalliday committed:
Exercise 2

Working Exercise 2 upload

DvdHalliday committed:
Exercise 3

Working Exercise 3 upload, and hence a .sln update

DvdHalliday committed:
Exercise 4

Working Exercise 3 upload, and current .sln file

JDAlzate commented on c++ training/W2S1/E2/Source.cpp:
```diff
@@ -0,0 +1,62 @@
+		std::cout << addOptionalZero(m_hours) << m_hours << addOptionalZero(m_minutes) << m_minutes << std::endl;
+	}
+
+	std::string addOptionalZero(short input) {
```
Your code makes sense and it's good. However, take a look at this method. It looks descriptive and follows the 'start with a verb' convention. However, it's not as descriptive as it looks. 

It's returning a string, but it doesn't make it clear what the string means. What does add zero mean and is it really optional? 

What are we doing then? We're basically taking a number and converting it to a string. We're also making sure that it is at least two digits. Finally, we're _getting_ a value, we're not actually adding anything to anything,, the function just returns new data. So a much clearer name would be something like `GetAsTwoDigitString()` or similar.

It's clear that we're returning a string based on the number parameter and it's clear that we are formatting to always follow two digits. What do you think?

DvdHalliday replied:
Yup it's confusing... ummm... the thing is.. I was avoiding string concatenation, so what that function actually does is just get a string that is either a 0 or an empty string, that's why it's "optional".  It doesn't actually return the value of the input, it's in fact being added in line 45 after the optional 0 gets printed, so I'm not entirely sure how to proceed here.

JDAlzate commented on c++ training/W2S1/E3/Source.cpp:
```diff
@@ -0,0 +1,93 @@
+	Circle(float radius) {
+		m_radius = radius;
+	}
+	float GetArea() {
```
I do realize that the exercises don't have any information about const, but it does instruct you to use const correctness. This is my mistake and will be soon fixed, but please as soon as you get here tomorrow, let me know and I'll give you a short explanation about const, because we do need to use it. Then, you can come back to this review and apply const correctness :)

JDAlzate commented on c++ training/W2S1/E3/Source.cpp:
```diff
@@ -0,0 +1,93 @@
+		return 3.14159265358979323846f *m_radius * m_radius;
+	}
+	float GetPerimeter() {
+		return 2 * 3.14159265358979323846f * m_radius;
```
Nice use of the f for the PI float, but you're using 2 as an integer and multiplying it by a float. Why not make it a float from the beginning?

JDAlzate commented on c++ training/W2S1/E3/Source.cpp:
```diff
@@ -0,0 +1,93 @@
+		return m_sideLength * m_sideLength;
+	}
+	float GetPerimeter() {
+		return 4 * m_sideLength;
```
Same here as previous comment.

JDAlzate commented on c++ training/W2S1/E4/Source.cpp:
```diff
@@ -0,0 +1,67 @@
+class Printer {
+private:
+	char m_outputString[60];
+	void Filler(const char newString[]) {
```
Remember that methods should start with verbs to communicate an action. Filler is a noun and is more suitable for an object, rather than a function.

JDAlzate reviewed (changes requested):
Review is up!

DvdHalliday committed:
First review's update

-Fixed the usage of integers in exercise to, converted them to floating points.
-Changed the name of Filler function to Fill so it is now a verb.
-I'm still missing some improvements in exercise 2 and exercise 3 so the latter is still a work in progress.

DvdHalliday committed:
Const correctness & method name

-Added const correctness in all projects that needed it ( Ex.1 doesn't)
-Changed that one confusing method name in Exercise 2

DvdHalliday committed:
Modified method and behaviour for exercise 2

Changed the function to return the string needed in the 4 digits format

JDAlzate reviewed (approved):
Nice! Approved!
